src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,446 @@
+import { JSONObject, validatePayload } from "../util/misc";
+import { ValueMetadata } from "../values/Metadata";
+import type { Maybe } from "../values/Primitive";
+import { CCAPI } from "./API";
+import {
+	API,
+	CCCommand,
+	CCCommandOptions,
+	ccValue,
+	ccValueMetadata,
+	commandClass,
+	CommandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+import { CommandClasses } from "./CommandClasses";
+
+// All the supported commands
+export enum ColorSwitchCommand {
+	SupportedGet = 0x01,
+	SupportedReport = 0x02,
+	Get = 0x03,
+	Report = 0x04,
+	Set = 0x05,
+}
+
+export enum ColorComponent {
+	WarmWhite = 0,
+	ColdWhite = 1,
+	Red = 2,
+	Green = 3,
+	Blue = 4,
+	Amber = 5,
+	Cyan = 6,
+	Purple = 7,
+	Index = 8,
+}
+
+export interface ColorTable {
+	warmWhite?: number;
+	coldWhite?: number;
+	red?: number;
+	green?: number;
+	blue?: number;
+	amber?: number;
+	cyan?: number;
+	purple?: number;
+	index?: number;
+}
+
+const ColorTableComponentMap: Record<keyof ColorTable, ColorComponent> = {
+	warmWhite: ColorComponent.WarmWhite,
+	coldWhite: ColorComponent.ColdWhite,
+	red: ColorComponent.Red,
+	green: ColorComponent.Green,
+	blue: ColorComponent.Blue,
+	amber: ColorComponent.Amber,
+	cyan: ColorComponent.Cyan,
+	purple: ColorComponent.Purple,
+	index: ColorComponent.Index,
+};
+
+@API(CommandClasses["Color Switch"])
+export class ColorSwitchCCAPI extends CCAPI {
+	public supportsCommand(cmd: ColorSwitchCommand): Maybe<boolean> {
+		switch (cmd) {
+			case ColorSwitchCommand.SupportedGet:
+			case ColorSwitchCommand.Get:
+			case ColorSwitchCommand.Set:
+				return true; // This is mandatory
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	public async supportedGet() {
+		this.assertSupportsCommand(
+			ColorSwitchCommand,
+			ColorSwitchCommand.SupportedGet,
+		);
+
+		const cc = new ColorSwitchCCSupportedGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = (await this.driver.sendCommand<
+			ColorSwitchCCSupportedReport
+		>(cc))!;
+		return {
+			supportsWarmWhite: response.supportsWarmWhite,
+			supportsColdWhite: response.supportsColdWhite,
+			supportsRed: response.supportsRed,
+			supportsGreen: response.supportsGreen,
+			supportsBlue: response.supportsBlue,
+			supportsAmber: response.supportsAmber,
+			supportsCyan: response.supportsCyan,
+			supportsPurple: response.supportsPurple,
+			supportsIndex: response.supportsIndex,
+		};
+	}
+
+	public async get(colorComponent: ColorComponent) {
+		this.assertSupportsCommand(ColorSwitchCommand, ColorSwitchCommand.Get);
+
+		const cc = new ColorSwitchCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent: colorComponent,
+		});
+		const response = (await this.driver.sendCommand<ColorSwitchCCReport>(
+			cc,
+		))!;
+		return {
```

Comment:
Later versions of Color Switch also return targets and durations, so we should at least keep this in object form.  It will make more sense once I get to color v3

Reply:
v3 is now implemented, so get now returns targetValue and duration in addition to currentValue and colorComponent